stellar/txobserver/main.go

```diff
@@ -226,37 +225,92 @@ func (o *Observer) processSingleLedger(ledger horizon.Ledger) error {
 	return nil
 }
 
-func (o *Observer) processOpsSinglePage(ops []operations.Operation, previousHash string) error {
-	for _, op := range ops {
-		baseOp := op.GetBase()
+func validTransaction(horizonTx *horizon.Transaction) bool {
+	// ignore failed transactions
+	if !horizonTx.Successful {
+		return false
+	}
+	gtx, err := txnbuild.TransactionFromXDR(horizonTx.EnvelopeXdr)
+	if err != nil {
+		return false
+	}
+	tx, ok := gtx.Transaction()
+	if !ok {
+		feeBump, _ := gtx.FeeBump()
+		tx = feeBump.InnerTransaction()
+	}
+	// Skip inserting transactions with multiple ops. Currently Starbridge
+	// does not create such transactions but it can change in the future.
+	return len(tx.Operations()) == 1
```

Comment:
The reason I chose to not validate the transaction memo for incoming payments is that validation of those fields should only be implemented in the HTTP handler for the withdrawal endpoint. If a user does not set the memo correctly on their payment to the bridge then the withdrawal should fail, however, it should still be possible to refund the payment. This will allow people who didn't set the memo properly to still be able to recover their deposit via refunds.